staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1293,33 +1293,34 @@ func (c *cacheWatcher) process(ctx context.Context, initEvents []*watchCacheEven
 
 type ready struct {
 	ok bool
+	m  *sync.RWMutex
```

Comment:
I don't see a reason for complicating this code - do you have any bencharmk showing it helps?

Reply:
I was looking at watch_cache_test and cacher_whitebox_test for test which can be transformed into benchmark.

cacher.ready.wait() is only called at the beginning of the test (which doesn't seem suitable being called in a loop).

Reply:
I don't think any of those are directly transformable to benchmark - if we want to prove it's usefulness this would have to simulated load on 5k-node cluster or something like that.

Anyway - I don't believe you would be able to prove it :)